File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, h: List[int]) -> int:
        n = len(h)
        i, j = 0, n - 1
        ans = 0
        while i < j:
            x = min(h[i], h[j])
            ans = max(ans, x * (j - i))
            if h[i] > h[j]:
                j -= 1
            else:
                i += 1
        return ans

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& h){
        int n = h.size();
        int i = 0, j = n - 1;
        int ans = 0;
        while(i < j){
            int x = min(h[i], h[j]);
            ans = max(ans, x * (j - i));
            if(h[i] > h[j]){
                j--;
            }
            else{
                i++;
            }
        }
        return ans;
    }
};